### ISSUE TITLE
Cannot reference classes or packages in module documentation file

### ISSUE THREAD

**Issue by IngoHaumann-TomTom** [Kotlin/dokka#1863]
When adding a `modules.md` file, we'd like to have links to certain packages that we're mentioning in that file. We can include the Markdown file without issue, but cannot seem to find the right way to mark up the links.

We would expect that `[com.common.target]` would lead us to the `index.html` of that package, as we are including all modules in our `DokkaTaskPartial` Gradle tasks, but it does not. We've tried the following way of writing those links but none of them worked:
```
- [Cannot reference a module by name](core.common.target)
- [Cannot reference a module by name with underscores](core_common_target)
- [Cannot reference a file, assuming we're at the `index.html` directory](./-my-class/-my-class.html)
- [Cannot reference a file, assuming we're at the root of all modules](./core_common_target/core_common_target/-my-class/-my-class.html)
- [Cannot reference a FQN for a class](com.example.core.common.target.MyClass)
- [Cannot reference using a relative path, assuming we're at the `index.html` directory](../../../../core_common_target/core_common_target/com.example.core.common.target/-my-class/index.html)
- [com.example.common.target.MyClass] Also cannot let Dokka create the link automatically.
```

They usually result in the link appearing verbatim in the output, while the anchor text is removed.

Ideally, when collecting a bunch of modules to create one documentation tree for the end user, you'd be able to easily reference anything in that collection. Using `\[square brackets\]` is the most intuitive.

Since this is not working, I consider this to be a bug.

(We're using 1.4.30 btw.)

**Comment by DanielRibeiroMaciel-TomTom** [comment]
Are there any workarounds?

**Comment by MarcinAman** [comment]
Well your solution assumes that package name is unique in the project, which is sometimes not true for multimodule projects. They would be unique if we prefix them with module name (and probably a sourceset?). But thats a solution up to discussion since it isn't a standard way kdoc links work

If you know where you will be hosting your documentation the easiest way is to add an external link there (a fully qualified one, like `https://www.google.com/`). This way after you upload new artefacts you will get correct links.

**Comment by MarcinAman** [comment]
As for the relative link to html: they don't work by design since linking to relative html paths creates a lot of issues when it comes to dokka model (pages can be relocated by plugins, you are limiting yourself to only one format)

**Comment by DanielRibeiroMaciel-TomTom** [comment]
If the fully qualified name is not unique (eg: 'foo.bar.mypackage.stuff.FooBar' is present in two modules), you could issue a warning and link to the first hit. The common use case is that if you are linking to it, then it is unique. It seems weird to force a fully qualified name when it can be disambiguated easily.
For instance, one could even just use the classname without the package name if that name is unique across the project.
Are there any technical limitations to that?

**Comment by IngoHaumann-TomTom** [comment]
Dokka has successfully created a repository of unique types of several modules that I can query as a human. It does not make sense that I cannot use that same repository to reference _any_ element in there, even it that elements lives in another module.

I can understand that a class name alone may prove difficult, but a package name added to it is a great disambiguator; just like Kotlin and Java are doing. If I reference something that is not unique, just throw an error so I can fix it.

**Comment by rnett** [comment]
This would be very useful for me, especially linking to things in other modules, linking to modules, and linking from the all modules page, all of which requires some sort of module qualification.  You would have to change the KDoc format for that, but as far as I know the module description format isn't super well defined anyways, all the [docs](https://kotlinlang.org/docs/kotlin-doc.html#module-and-package-documentation) say is "a Markdown file".

**Comment by IngoHaumann-TomTom** [comment]
It looks like I cannot add a `bug` label, so I wonder if this issue will be considered as such.

Could somebody with more editing rights please add it? Thank you very much.

**Comment by IngoHaumann-TomTom** [comment]
@MarcinAman, could you please add the `bug` label? Or are you waiving this issue?

**Comment by strangesource** [comment]
Any ETA for this? This would be an awesome feature that our documentation would really profit from. 🙂 

Or did anyone come up with a workaround by now?

**Comment by something15525** [comment]
Agreed, this would be a great addition my documentation sorely needs!

**Comment by frne** [comment]
Any update on this? I just came along (using `2.0.0-Beta`) and noticed, beyond the issue mentioned by @IngoHaumann-TomTom, relative links are not working at all.

We have knit TOCs in our README's that produces structures like the following (which is perfectly valid Markdown):

```markdown
<!--- TOC -->

* [Usage](#usage)
* [Installation](#installation)
* [Dev-Setup](#dev-setup)

<!--- END -->

# Usage
...

# Installation
...

# Dev-Setup
...
```

However, all the links are rendered as the target anchors. I'd expect dokka to generate valid HTML from valid Markdown, 
and would therefore also assume that this is a bug.

**Comment by rnett** [comment]
I would also like to be able to reference other modules.  My multi-module module documentation gives a brief overview of the library, and I would like to be able to link to the relevant modules from the body text, rather than relying purely on the list.  For example, something like `Provides integrations with [kotlinpoet](link to Kotlinpoet integration module) and [KSP](link to KSP integration module)`.
